src/components/code-editor/index.js

```diff
@@ -143,24 +134,22 @@ module.exports = {
 	    this.saving = true;
 	    var bytes = convertToByteArray(new TextEncoder().encode(text));
 	    var java_reader = peergos.shared.user.fs.AsyncReader.build(bytes);
-	    const file = this.currentFile;
 	    const context = this.context;
-	    const size = text.length;
+	    const size = this.currentFile.getFileProperties().sizeLow();
 	    const that = this;
-        this.currentParent.uploadFileJS(file.getName(), java_reader, (size - (size % Math.pow(2, 32)))/Math.pow(2, 32), size,
-                true, true, context.network, context.crypto, len => {}, context.getTransactionService())
-        .thenApply(function(updatedParent) {
-            updatedParent.getChild(file.getName(), context.crypto.hasher, context.network).thenApply(function(updatedFile) {
-                that.currentFile = updatedFile.get();
-                that.currentParent = updatedParent;
-                that.saving = false;
-            });
+
+        this.currentFile.overwriteSectionJS(java_reader, 0, 0, (size - (size % Math.pow(2, 32)))/Math.pow(2, 32), size,
+            context.network, context.crypto, len => {})
+        .thenApply(function(updatedFile) {
+            that.saving = false;
+            that.currentFile = updatedFile;
         }).exceptionally(function(throwable) {
-            if (throwable.detailMessage.includes("java.lang.JsException: Couldn%27t+update+mutable+pointer%2C+cas+failed%3A+")) {
-                that.showMessage("Concurrent modification detected", "The file: '" + file.getName() + "' has been updated by another user. Your changes could not been saved.");
+            if (throwable.detailMessage.includes("CAS exception")) {
```

Comment:
we still need the previous cas error which is a different type of error

Reply:
It means something else in the same writing space was changed, so in principle we should be able to handle it automatically though.

Reply:
Actually yeah, let's leave it out, and assume we'll handle it internally, by retrying.